src/main/java/eu/dissco/backend/controller/UserController.java

```diff
@@ -0,0 +1,124 @@
+package eu.dissco.backend.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.dissco.backend.domain.JsonApiLinks;
+import eu.dissco.backend.domain.JsonApiWrapper;
+import eu.dissco.backend.exceptions.ConflictException;
+import eu.dissco.backend.exceptions.ForbiddenException;
+import eu.dissco.backend.exceptions.NotFoundException;
+import eu.dissco.backend.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.keycloak.KeycloakPrincipal;
+import org.keycloak.KeycloakSecurityContext;
+import org.keycloak.representations.AccessToken;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Slf4j
+@RestControllerAdvice
+@RestController
+@RequestMapping("/api/v1/users")
+@RequiredArgsConstructor
+public class UserController {
+
+  private static final String SELF_LINK = "https://sandbox.dissco.tech/api/v1/users/";
+
+  private final UserService service;
+
+  private static void checkAuthorisation(String tokenId, String id) throws ForbiddenException {
```

Comment:
Just to clarify: @PreAuthorize("isAuthenticated") checks if the user has permission to access the API at all, and the checkAuthorisation function checks the user's specific permissions for each action?

Reply:
Yes there is a hasRole as well to check if a user has a certain role and I believe a couple of others to check group.